install/kubernetes/cilium/charts/nodeinit/templates/daemonset.yaml

```diff
@@ -93,6 +93,21 @@ spec:
               if grep -q 'docker' /etc/crictl.yaml; then
                 # Works for COS, ubuntu
                 for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0`; do docker rm -f $(cat $f) || true; done
+              elif [ -n "$(docker ps | grep [0-9].dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni)" ]; then
```

Comment:
How much of this FQDN is dynamic? The `us-west-2` part looks dependent on which region you deploy your cluster, so presumably we would need to make that part variable at the very least.

Reply:
You're right, I generalized/improved the regex.